src/coreclr/jit/gentree.cpp

```diff
@@ -21592,11 +21592,55 @@ GenTree* Compiler::gtNewSimdBinOpNode(
                     {
                         intrinsic = NI_AVX10v1_MultiplyLow;
                     }
-                    else
+                    else if (compOpportunisticallyDependsOn(InstructionSet_AVX512DQ_VL))
```

Comment:
Now that we're doing a "proper" check here, we want to combine this and the previous path using:
```cpp
bool isV512Supported = false;
if (compIsEvexOpportunisticallySupported(&isV512Supported, InstructionSet_AVX512DQ_VL))
{
    if (simdSize == 64)
    {
        assert(isV512Supported);
        intrinsic = NI_AVX512DQ_MultiplyLow;
    }
    else
    {
        intrinsic = !isV512Supported ? NI_AVX10v1_MultiplyLow : NI_AVX512DQ_VL_MultiplyLow;
    }
}
```

The reasoning being that `Avx10v1` and `Avx512DQ_VL` are part of the same synthetic `EVEX` baseline and we want the capability of eventually tracking that this is an either/or scenario for R2R/AOT scenarios.

Reply:
Addressed